### PR TITLE
allow all numbers in jwt token type names

### PR DESCRIPTION
**Pull request by gkchestertron**
currently naming of the jwt_token type, as in the tutorial/example only allows for names that contain the numbers 1 and 2. I'm using this for multiple PTA organizations and would like to keep consistent naming and identify them by their district number (i.e. pta_dist_14.jwt_token). 

This fixes the issue - but I am unsure if there was a good reason that only 1 and two were allowed before.

### CODE REVIEW

**Comment by benjie** [comment]
O_o this seems weird; surely that'd have to be deliberate... @calebmer any idea?

**Comment by gkchestertron** [comment]
right? but I couldn't think of one.

**Comment by benjie** [comment]
Tracking through various levels of git blame; this is where it was originally introduced:

https://github.com/postgraphql/postgraphql/commit/f8b2d42c4d17dfd099a1762f28f33d675b50f219

> "Type identifiers are a bit tricky, but this function uses a regular expression to do it. Fun."

Worrisome... I wonder what's tricky about them.

I had a scan through the docs and can't see any issues:

https://www.postgresql.org/docs/9.6/static/sql-createtype.html

Could you add some tests so we can be certain?

**Comment by gkchestertron** [comment]
will do.

**Comment by gkchestertron** [comment]
Let me know if I've complete missed on your conventions - I'm new to ES6, TypeScript, Jest, Travis, and ESLint. Also - sorry this is another suite of slow tests in the postgraphql space.

File: src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import withPgClient from '../../../../__tests__/utils/withPgClient'
+import introspectDatabase from '../../../../postgres/introspection/introspectDatabase'
+import getPgTokenTypeFromIdentifier from '../getPgTokenTypeFromIdentifier'
+
+// good names start with a letter or underscore and may also contain numbers after position 0
+const goodNames = [
+  'test', 'test_0', 'test_1', 'test_2', 'test_3', 'test_4', 'test_5', 'test_6', 'test_7', 'test_8', 'test_9', '_test',
```
Comment (benjie) [inline src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js]: This only really needs to test edge cases, so 'test_0', 'test_9' and '_test' should suffice

File: src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    let catalog   = await introspectDatabase(client, [goodName])
+    let tokenType = getPgTokenTypeFromIdentifier(catalog, `${goodName}.jwt_token`)
+
+    expect(tokenType.name).toBe('jwt_token')
+  }))
+})
+
+// schema, type, etc... names are not allowed to start with numbers
+const badNames = [
+  '0_test', '1_test', '2_test', '3_test', '4_test', '5_test', '6_test', '7_test', '8_test', '9_test',
```
Comment (benjie) [inline src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js]: similarly I'd only test '0_test' and '9_test'

File: src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+]
+
+for (var i in badNames) {
+  let badName = badNames[i]
+
+  test(`Bad schema name ${badName} should throw a syntax error on create`, withPgClient(async client => {
+    let badQuery = 'not an error'
+
+    try {
+      badQuery = await client.query(`create schema ${badName}; create type ${badName}.jwt_token as (role text, exp integer, a integer, b integer, c integer)`)
```
Comment (benjie) [inline src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js]:
```
create schema "0_test";
CREATE SCHEMA
Time: 35.303 ms
```

File: src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js (hunk 1 above)
Comment (gkchestertron) [inline src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js]: fair enough

File: src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js (hunk 3 above)
Comment (gkchestertron) [inline src/postgraphql/schema/auth/__tests__/getPgTokenTypeFromIdentifier-test.js]: you're right. works fine when escaped.

**Comment by benjie** [comment]
@gkchestertron Are you good for this to be merged?

**Comment by gkchestertron** [comment]
works for me
